### PR TITLE
Issue #2 Update the README

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,131 @@
-# Minishift Fedora VM
+<a name="creating-a-minishift-fedora-iso"></a>
+# Minishift Fedora ISO
 
 This repository contains all the instructions and code to build a Live ISO based on Fedora
 which can be used by [minishift](https://github.com/minishift/minishift) as an alternative to
-the boot2docker and CentOS ISO.
+the boot2docker ISO.
```
Comment (budhrg) [inline README.md]: Isn't this should be CentIOS ISO ? This line is telling me that we have b2d ISO still as default.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,131 @@
...
+- [Community](#community)
+
+<!-- /MarkdownTOC -->
+
+----
+
+<a name="building-the-fedora-iso"></a>
+## Building the Fedora ISO
+
+The following contains instructions on how to build the default (Fedora based) ISO.
```
Comment (budhrg) [inline README.md]: `the default (Fedora based) ISO` is confusing me. Fedora ISO isn't our default ISO.

 Probably line should be
```
The following contains instructions on how to build the Fedora based ISO.
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,131 @@
...
+If you are able to install [livecd-tools](https://github.com/rhinstaller/livecd-tools)
+directly on your machine, you can use the [Fedora](#on-fedora) instructions.
+
+If you don't have _livecd-tools or using different linux distro other than fedora_, follow the
```
Comment (budhrg) [inline README.md]: `other than Fedora_`.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,131 @@
...
+#### Building the ISO
+
+```
+$ git clone https://github.com/minishift/minishift-fedora-iso.git
+$ cd minishift-fedora-iso
+$ make
+```
+
+<a name="non-fedora-hosts"></a>
+### On hosts _other than Fedora_ (OS X, Windows, CentOS ...)
```
Comment (budhrg) [inline README.md]: Let's remove `Windows`. We have never tested it as I remember.
Did you @praveenkumar @LalatenduMohanty @gbraad @agajdosi ?

If we can't sure, we shouldn't mention it.
Also, I am not interested in building ISO in Windows at all :smile:

Comment (budhrg) [reply README.md]: Change `OS X` to `macOS`.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,131 @@
...
+
+<a name="prerequisites-1"></a>
+#### Prerequisites
+
+* [Vagrant](https://www.vagrantup.com/)
```
Comment (budhrg) [inline README.md]: We might need to pin the Vagrant version as this is painful to see everytime user compalins this version not working etc etc. But for now it is OK IMO.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,131 @@
...
+  $ make release
+  ```
+
+#### Build ISO
+
+Setup your build environment by following the instructions provided in [Building the Fedora ISO](#building-the-fedora-iso) section as per your preferred OS.
+
+Note: Building ISO might require you to have Vagrant environment if you are not using host other than Fedora.
+
+#### Run the tests:
```
Comment (budhrg) [inline README.md]: No other header has `:` (Colon), let's remove it from here as well.

File: README.md [hunk 7]
```diff
@@ -1,5 +1,131 @@
...
+
+Once the ISO is built from above step, use following command to run tests:
+
+```
+$ make test
+```
+
+Note: If you are using the Vagrant environment, you need to exit from it and come back to host to run the above command.
+
+This command will fetch the latest [Minishift](http://github/minishift/minishift) binary and run the [tests](tests/test.sh).
```
Comment (budhrg) [inline README.md]: `This command will fetch the latest [minishift](http://github/minishift/minishift) binary...`

File: README.md (hunk 1 above)
Comment (gbraad) [reply README.md]: I read the same. We soon do not have b2d anymore...

File: README.md (hunk 4 above)
Comment (gbraad) [reply README.md]: Vagrant and Windows works well... I would leave it in

Comment (budhrg) [reply README.md]: Just want to be consistent here. :) Its difficult to compile binary in Windows.

Comment (gbraad) [reply README.md]: We are NOT building a binary here, but the ISO. and so far we have described each ISO to be able to be build using Vagrant. Besides, we might have to do this anyway, as on CentOS you have to use a VM to build the Fedora ISO... you can not rely on CentOS to provide a workable result.

Comment (gbraad) [reply README.md]: Besides, customization of the ISO is a possiblity that people want, and this has nothing to do with the need to build the actual minishift binary.

Comment (budhrg) [reply README.md]: @gbraad , If it is working in Windows then I am fine :)

**Pull request merged**